### PR TITLE
Write custom catalog yaml from simple importer.

### PR DESCRIPTION
**Pull request by keyurva**
* This PR is the result of a discussion to have the simple importer write the catalog yaml.
* Doing so allows both the build embeddings tool and the nl server to generate and load embeddings respectively with the same catalog path.
* This replaces similar logic that is currently being done in the nl server's `admin/html.py`. This follow-on PR removes that logic from there: https://github.com/datacommonsorg/website/pull/4473

### CODE REVIEW

File: simple/stats/nl.py [hunk 1]
```diff
@@ -53,9 +65,58 @@ def generate_nl_sentences(triples: list[Triple], sentences_fh: FileHandler):
...
+              "type": "LOCAL",
+              "usage": "EMBEDDINGS",
+              "gcs_folder": _CUSTOM_MODEL_PATH,
+              "score_threshold": 0.5
+          }
+      }
+  }
+
+
+def _rewrite_catalog_for_testing(catalog_fh: FileHandler):
```
Comment (keyurva) [inline simple/stats/nl.py:105]: Don't like this approach but can't think of an alternative. Let me know if any suggestions.

File: simple/tests/stats/nl_test.py [hunk 2]
```diff
@@ -38,20 +38,27 @@ def _test_generate_nl_sentences(test: unittest.TestCase, test_name: str):
...
+    output_catalog_yaml_path = os.path.join(temp_dir, "embeddings",
+                                            "custom_catalog.yaml")
+    expected_catalog_yaml_path = os.path.join(_EXPECTED_DIR, test_name,
+                                              "custom_catalog.yaml")
 
-    nl.generate_nl_sentences(input_triples, output_sentences_csv_fh)
+    nl_dir_fh = LocalFileHandler(temp_dir)
+
+    nl.generate_nl_sentences(input_triples, nl_dir_fh)
+    nl._rewrite_catalog_for_testing(LocalFileHandler(output_catalog_yaml_path))
```
Comment (pradh) [inline simple/tests/stats/nl_test.py:53]: The path we are rewriting/replacing is "temp_dir" right?  So should we just str-replace "temp_dir" with fake-path here, and not have any test-specific code in the main files?

File: simple/stats/nl.py (hunk 1 above)
Comment (pradh) [reply simple/stats/nl.py:105]: Left a suggestion

**Review by pradh — COMMENTED** [review]
Maybe update the description with some context from discussion with Bo?

Is this going to subsume the admin.py logic that also does similar things?

File: simple/stats/nl.py [hunk 3]
```diff
@@ -20,13 +20,25 @@
 from stats.data import Triple
 import stats.schema_constants as sc
 from util.filehandler import FileHandler
+import yaml
 
 _DCID_COL = "dcid"
 _SENTENCE_COL = "sentence"
 _SENTENCE_SEPARATOR = ";"
 
+_CUSTOM_EMBEDDING_INDEX = "user_all_minilm_mem"
```
[the remark this answers is not in the document]
Comment (keyurva) [reply simple/stats/nl.py]: Moved them to a separate file with comments.

File: simple/stats/nl.py [hunk 4]
```diff
@@ -53,9 +65,58 @@ def generate_nl_sentences(triples: list[Triple], sentences_fh: FileHandler):
 
   dataframe = pd.DataFrame(rows)
 
+  sentences_fh = nl_dir_fh.make_file(_SENTENCES_FILE)
   logging.info("Writing %s NL sentences to: %s", dataframe.size, sentences_fh)
   sentences_fh.write_string(dataframe.to_csv(index=False))
 
+  embeddings_dir_fh = nl_dir_fh.make_file(f"{_EMBEDDINGS_DIR}/")
```
[the remark this answers is not in the document]
Comment (keyurva) [reply simple/stats/nl.py:70]: Added comment.

File: simple/stats/nl.py (hunk 1 above)
Comment (keyurva) [reply simple/stats/nl.py]: Thanks.

File: simple/stats/nl.py [hunk 5]
```diff
@@ -53,9 +65,58 @@ def generate_nl_sentences(triples: list[Triple], sentences_fh: FileHandler):
...
+  """
+  Test catalog files are written to temp folders whose values change from test to test.
+  To consistently test the catalog out against a golden file, we replace the root of paths in the catalog
+  with a constant fake path.
+  """
+  catalog_dict = yaml.safe_load(catalog_fh.read_string())
+  custom_index = catalog_dict["indexes"][_CUSTOM_EMBEDDING_INDEX]
+  orig_source_path = custom_index["source_path"]
+  custom_index["source_path"] = _FAKE_SOURCE_PATH
+  orig_embeddings_path: str = custom_index["embeddings_path"]
```
[the remark this answers is not in the document]
Comment (keyurva) [reply simple/stats/nl.py]: This function is now gone.

File: simple/tests/stats/nl_test.py (hunk 2 above)
Comment (keyurva) [reply simple/tests/stats/nl_test.py]: That is a great suggestion - thanks. Done.

File: simple/stats/nl_constants.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+generating a custom catalog yaml.
+
+See this for full reference:
+
+https://github.com/datacommonsorg/website/blob/1f46ef4f22d4777194f4dc6bb730ea56ca6f3207/deploy/nl/catalog.yaml#L119-L123
+"""
+
+# The custom embeddings index type.
+# See: https://github.com/datacommonsorg/website/blob/2dabf815b82dcd5ff226ed94d8bbdab545ffd4d7/nl_server/custom_dc_env.yaml#L6
+CUSTOM_EMBEDDINGS_INDEX = "user_all_minilm_mem"
```
Comment (pradh) [inline simple/stats/nl_constants.py:25]: super nit: some vertical space between variable and start of next multiline comment block

**Review by pradh — APPROVED** [review]
Thanks for the changes!

File: simple/stats/nl_constants.py (hunk 6 above)
Comment (keyurva) [reply simple/stats/nl_constants.py:25]: Done.

**Review by keyurva — COMMENTED** [review]
Thanks for the review!

**Pull request merged**